### PR TITLE
Overhaul docker image for cloud

### PR DESCRIPTION
**Pull request by gbbirkisson**
## Noteable changes

* Docker image overhauled
    * Uses `enonic-xp-generic` distro
    * Uses `adoptopenjdk:11-jre-hotspot` as base image
    * Include [Jattach](https://github.com/apangin/jattach) for easy heap and thread dumps
    * Sets locale (Issue when locale not set and you import node exports)
    * Set labels according to [label-schema](http://label-schema.org/rc1/)
    * Set labels according to [opencontainers](https://github.com/opencontainers/image-spec/blob/master/annotations.md)
    * Add [openshift support](https://access.redhat.com/documentation/en-us/openshift_container_platform/3.10/html/creating_images/creating-images-guidelines)
    * Create 'standard' dirs so docker mounts them with correct file permissions:
        * $XP_HOME/config
        * $XP_HOME/data
        * $XP_HOME/deploy
        * $XP_HOME/logs
        * $XP_HOME/repo/blob
        * $XP_HOME/repo/index
        * $XP_HOME/snapshots
        * $XP_HOME/work
* Includes more scripts
    * `app.sh` to easly install apps into deploy folder
    * `backups.sh` to check snapshot age
    * `dump.sh` to create heap and thread dumps
    * `docker-entrypoint.sh` Setup environment and user before running XP ([inspired by the official elastic search image](https://github.com/elastic/elasticsearch/blob/master/distribution/docker/src/docker/bin/docker-entrypoint.sh))
* Include `docker-compose` file for easy local debugging
* Build script altered
    * Always pull latest base image
    * Do not use cache to always get latest versions when installing from apt-get

## Breaking changes

This image has 2 breaking changes, they are.

#### Folder $XP_ROOT/home.org is just called $XP_ROOT/home

The old docker image for some reason moved `$XP_ROOT/home` to `$XP_ROOT/home.org` and then made the `launcher.sh` script copy it back. In EC we copy it back during the build process. I don`t understand all this copying back and forth. Now the distro is just put in the image unmodified.

This means that in EC, all builds doing `RUN cp -r $XP_ROOT/home.org $XP_ROOT/home` will fail. The fix is as simple as removing that statement in the EC build process.

#### wget is not in the image

The image uses `adoptopenjdk:11-jre-hotspot` as base image that comes with `curl` preinstalled. I just did not see any point in adding `wget` in there aswell.

We have 2 options here. Either just install `wget` during builds in EC or add `wget` to the image.

### CODE REVIEW

**Comment by gbbirkisson** [comment]
@diegopasten Weren't these the only steps to migrate the Dockerfiles in EC to this one?

**Comment by rymsha** [comment]
> Includes more scripts

Could some (most) of them be part of xp-distro?

File: .env [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+XP_VERSION=7.2.1
+
+IMAGE_NAME="enonic/xp"
+IMAGE_TAG="${XP_VERSION}-ubuntu"
+IMAGE_FULL_TAG="${IMAGE_NAME}:${IMAGE_TAG}"
+
+DOCKER_BUILD_EXTRA_ARGS="--no-cache --pull"
+
+PUSH_TO="docker.io/$IMAGE_FULL_TAG"
+#PUSH_TO="docker.io/$IMAGE_FULL_TAG gcr.io/public-resources-235813/$IMAGE_FULL_TAG"
```
Comment (rymsha) [inline .env:12]: Do we need this commented out line (usually repo name is a secret, although I don't know why)

File: docker-compose.yml [hunk 2]
```diff
@@ -1,17 +1,48 @@
...
+        limits:
+          cpus: "2.00"
+          memory: "1024M"
     ports:
+      - "1099:1099"
+      - "2609:2609"
+      - "3000:3000" # Profiling test
+      - "3001:3001" # Profiling test
+      - "4848:4848"
+      - "5005:5005" # Remote debugging test
```
Comment (rymsha) [inline docker-compose.yml:22]: 3000 3001 and 5005 look open just for testing. Is this file for testing only?

**Review by rymsha — APPROVED** [review]

File: docker-compose.yml (hunk 2 above)
Comment (gbbirkisson) [reply docker-compose.yml:22]: Yes the docker-compose file is only for local testing while developing the image. This is not the default behavior when it is used.

File: .env (hunk 1 above)
Comment (gbbirkisson) [reply .env:12]: I do not think we need it but it was in the old script, so I kept it in. Maybe @diegopasten knows why it was there in the first place.

**Comment by gbbirkisson** [comment]
> > Includes more scripts
> 
> Could some (most) of them be part of xp-distro?

I don`t think we should pollute the distro with them:
* `app.sh` This one you could argue should be a part of the distro, but it is just a simple wrapper for curl, and most people just install apps in the admin anyway.
* `backups.sh` Only relevant when the snapshotter app is up and running.
* `dump.sh` Requires `jattach` so would not work on most systems.
* `docker-entrypoint.sh` Docker specific script, should not be in the distro.

They are also written for unix, you would probably have to do some trickery to use them on windows.

**Review by jsi — APPROVED** [review]
I have created an image of 7.2.3 locally, using this recipe, and it works great.  Lots of small details that I love, like `ll` working.  Have not done any load testing or other heavy work with this image though.

**Review by runarmyklebust — APPROVED** [review]

**Review by diegopasten — APPROVED** [review]

**Pull request merged**